### DISCUSSION TITLE
Read Servlet Init-Params from IAS4ServletMessageProcessorSPI implementation

### DISCUSSION THREAD

**Discussion by binaradarsha** [phax/phase4#184] [General]
Hi,

I need to read a servlet init-param from `IAS4ServletMessageProcessorSPI` implementation class. Since theoretically it cannot be done directly in `IAS4ServletMessageProcessorSPI`, first I read it from our `AbstractXServlet` implementation and finding a way to set in some parameter of `IAS4ServletMessageProcessorSPI.processAS4UserMessage`, i.e. to `IAS4IncomingMessageMetadata`.
But still no luck. Could you please show some path?

**Comment by phax** [comment]
In your `AS4XServletHandler` implementation, add this method to capture the Servlet Init-parameters:

```java
 public void onServletInit (ICommonsMap <String, String> aInitParams) throws ServletException
  {
    // TODO
  }
```
It doesn't really fit into the `IAS4IncomingMessageMetadata`, because this data is meant for "per transmission" data only.
hth

**Reply by binaradarsha** [reply]
Hi Philip,
My intention was to configure two urls in our AS4 inbound service, one for prod (/as4) and other for test (/as4-test). 

Initially I was planning to add two `<servlet> & <servlet-mapping>` parts in web.xml with different `<init-param>` to mention whether it is test or prod endpoint. So based on the url/endpoint a document arrived it expected to carry that `<init-param>` value in `IAS4IncomingMessageMetadata`. But I couldn't see init params are available in `IAS4IncomingMessageMetadata`.

Therefore I added two <url-pattern> tags in same endpoint config. Then using `AS4XServletHandler.setHandlerCustomizer()` I set servletPath to request headers like this, and access in `IAS4ServletMessageProcessorSPI.processAS4UserMessage`.
`aRequestScope.headers().setHeader("ServletPath", aRequestScope.getServletPath())` 

Not sure this is a commended way, but this setup works.
